app/com/arpnetworking/metrics/portal/scheduling/JobExecutorActor.java

```diff
@@ -96,42 +100,108 @@ public void preStart() throws Exception {
      * @param wakeUpBy The time we want to be sure we wake up by.
      * @return The time until we should wake up. Always between 0 and (approximately) {@code wakeUpBy}.
      */
-    /* package private */ FiniteDuration timeUntilNextTick(final Instant wakeUpBy) {
+    /* package private */ Optional<FiniteDuration> timeUntilExtraTick(final Instant wakeUpBy) {
         final FiniteDuration delta = Duration.fromNanos(ChronoUnit.NANOS.between(_clock.instant(), wakeUpBy));
         if (delta.lt(Duration.Zero())) {
-            return Duration.Zero();
+            return Optional.of(Duration.Zero());
         } else if (delta.lt(TICK_INTERVAL)) {
-            return delta;
-        } else {
-            return TICK_INTERVAL;
+            return Optional.of(delta);
+        }
+        return Optional.empty();
+    }
+
+    private void scheduleExtraTickIfNecessary(final Instant wakeUpBy) {
+        timeUntilExtraTick(wakeUpBy).ifPresent(timeRemaining -> {
+            timers().startSingleTimer(EXTRA_TICK_TIMER_NAME, Tick.INSTANCE, timeRemaining);
+        });
+    }
+
+    private void killSelf() {
+        // TODO(spencerpearson): Per https://doc.akka.io/docs/akka/2.5.4/java/cluster-sharding.html#remembering-entities ,
+        //   might we need to send a Passivate message to our parent?
```

Comment:
The simple policy is to always send the passivate and be removed from the cluster (once you add that functionality). 

However, this policy can be optimized (for background) based on the case you that you are shutting down. In the case the job does not exist you want to send the passivate and not be resurrected. In the case that there is a transient error you probably don't mind getting resurrected elsewhere. Although we don't have retries the job will continue to execute a scheduled run until it is able to record failure for that run (so it the face of meta-failures we try forever). Removing from the cluster in this case (and any other where the job still exists) doesn't matter because the JobCoordinator will resurrect the job anyway.

Anyway, doesn't apply until the next (?) review when you add cluster registration of these actors.